UnitySetup/UnitySetup.psm1

```diff
@@ -2132,3 +2130,537 @@ function Get-UnityLicense {
         Write-Warning "Alias $($_.Name) already configured by $($alias.Source)"
     }
 }
+
+function Import-UPMConfig {
+    [CmdletBinding()]
+    param(
+        [Parameter()]
+        [String]$ProjectManifestPath,
+        [int]$SearchDepth = 3
+    )
+
+    $ProjectManifestPaths = @()
+
+    if ((Get-Item $ProjectManifestPath) -is [System.IO.DirectoryInfo]) {
+        Write-Host "Path provided is not a manifest.json file ($ProjectManifestPath), will attempt search"
+        $FoundPaths = @(Get-Childitem -Path $ProjectManifestPath -Include manifest.json -File -Recurse -Depth $SearchDepth -ErrorAction SilentlyContinue)
+        foreach ($file in $FoundPaths) {
+            $ProjectManifestPaths += $file.FullName
+            if ($Verbose) { Write-Host "Found ($file.FullName)" }
+        }
+    }
+    else {
+        $ProjectManifestPaths += $ProjectManifestPath
+    }
+
+    if (([string]::IsNullOrEmpty($ProjectManifestPath)) -or (-not $(Test-Path $ProjectManifestPath))) {
+        Write-Error "Unable to find manifest.json file, please provide a path pointing directly to a Unity project's manifest.json or provide a path to a Unity project"
```

Comment:
Nit: I think I would throw here rather than write-error and return.

Reply:
Updated